### PR TITLE
feat(TX-242): pay by saved bank account 

### PR DESCRIPTION
**Pull request by rquartararo**
The type of this PR is: **Feature**

This PR solves [TX-242]

### Description

This PR adds the ability for a user to pay for an order with a bank account that was previously used. **Note:** this does not include the "save account for later user" checkbox, so we are currently displaying all bank accounts that have been added to an order previously or during checkout. 

### Screen recording

https://user-images.githubusercontent.com/50849237/175008891-00745312-692a-49fe-81e1-bd237b295a20.mov

[GRO-434]: https://artsyproduct.atlassian.net/browse/GRO-434?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[TX-242]: https://artsyproduct.atlassian.net/browse/TX-242?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/v2/Apps/Order/Components/BankAccountPicker.tsx [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+  const { submitMutation: setPaymentMutation } = useSetPayment()
+
+  const handleContinue = async () => {
+    try {
+      const orderOrError = (
+        await setPaymentMutation({
+          variables: {
+            input: {
+              id: order.internalID,
+              paymentMethod: "US_BANK_ACCOUNT",
```
[the remark this answers is not in the document]
Comment (fladson) [reply src/v2/Apps/Order/Components/BankAccountPicker.tsx:64]: Yes, `US_BANK_ACCOUNT` should be used when the account already has a `bank_account_id` and `ACH_TRANSFER` should be used when a new intent is requested by the user.

File: src/v2/Apps/Order/Components/BankAccountPicker.tsx [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+export const BankAccountPicker: FC<Props> = props => {
+  const getInitialBankAccountSelection = () => {
+    if (props.order.bankAccountId) {
+      return {
+        type: "existing",
+        id: props.order.bankAccountId,
+      }
+    } else {
+      return props.me.bankAccounts?.edges?.length
```
Comment (starsirius) [inline src/v2/Apps/Order/Components/BankAccountPicker.tsx:31]: #non-blocking. This can probably reuse `userHasExistingBankAccounts` below.

File: src/v2/Apps/Order/Components/BankAccountPicker.tsx [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+        ? {
+            type: "existing",
+            id: props.me.bankAccounts.edges[0]?.node?.internalID!,
```
Comment (starsirius) [inline src/v2/Apps/Order/Components/BankAccountPicker.tsx:34]: #non-blocking. Similarly, maybe reusing `bankAccountsArray`?

File: src/v2/Apps/Order/Components/BankAccountPicker.tsx [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+              >
+                Add another bank account.
+              </BorderedRadio>,
+            ])}
+        </RadioGroup>
+      )}
+      <Spacer mb={4} />
+      <Collapse
+        open={
+          bankAccountSelection.type === "new" || !userHasExistingBankAccounts
```
Comment (starsirius) [inline src/v2/Apps/Order/Components/BankAccountPicker.tsx:123]: Perhaps `userHasExistingBankAccounts` is redundant if we set the initial bank account selection properly.

File: src/v2/Apps/Order/Routes/Payment/PaymentContent.tsx [hunk 5]
```diff
@@ -70,7 +74,7 @@ export const PaymentContent: FC<Props> = props => {
       <PaymentContentWrapper isLoading={isLoading}>
         <PaymentPickerFragmentContainer
```
[the remark this answers is not in the document]
Comment (starsirius) [reply src/v2/Apps/Order/Routes/Payment/PaymentContent.tsx:75]: 💯  supporting this!

File: src/v2/Apps/Order/Components/BankAccountPicker.tsx [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+    } catch (error) {
+      onSetPaymentError(error)
+    }
+  }
+
+  return (
+    <>
+      {userHasExistingBankAccounts && (
+        <RadioGroup
+          onSelect={val => {
```
Comment (starsirius) [inline src/v2/Apps/Order/Components/BankAccountPicker.tsx:85]: #non-blocking. I feel the rendering here is a bit hard to follow because of all the inline logic. Perhaps it reads better if we move the logic to some functions and the rendering would look more concise.

**Review by kiryl-zubarau — APPROVED** [review]
LGTM! I have just a few non-blocking comments
